scaffold/github/workflows/ComposerLockDiff.yml

```diff
@@ -17,46 +17,33 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 2
+          fetch-depth: 0
 
-      - uses: actions/cache@v4
+      - name: Check if composer.lock was changed
+        id: composer-lock-changed
+        uses: tj-actions/changed-files@v44
         with:
-          path: ${{ github.workspace }}/.ddev/.drainpipe-composer-cache
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-composer-
+          files: 'composer.lock'
 
-      - uses: ./.github/actions/drainpipe/set-env
-
-      - name: Install and Start DDEV
-        uses: ./.github/actions/drainpipe/ddev
+      - name: Delete sticky pull request comment
+        uses: marocchino/sticky-pull-request-comment@v2
         with:
-          git-name: Drainpipe Bot
-          git-email: no-reply@example.com
-          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
-          ssh-known-hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
-
-      - name: Build Project
-        run: ddev composer install
-
-      - name: Install composer lock diff
-        run: ddev composer global require davidrjonas/composer-lock-diff:^1.0
+          header: composer-lock-diff
+          delete: true
 
-      - name: Prepare repository
-        run: |
-          git reset --soft HEAD^1
-          git reset .
+      - name: Generate composer diff
+        if: ${{ steps.composer-lock-changed.outputs.any_changed == 'true' }}
+        id: composer-diff
+        uses: IonBazan/composer-diff-action@v1
+        with:
+          with-platform: true
+          with-links: true
 
-      - name: Run composer lock diff
-        run: |
-          curl -f \
-            -H "Accept: application/vnd.github+json" \
-            -H "Authorization: Bearer  ${{ secrets.GITHUB_TOKEN }}" \
-            https://api.github.com/repos/${{ github.repository }}/pulls/$DRAINPIPE_PR_NUMBER | jq '. | {body}' > pull_request.json
-          ddev task "github:composer-lock-diff pull_request='pull_request.json' json_file='processed.json'"
-          curl -f \
-            -X PATCH \
-            -H "Accept: application/vnd.github+json" \
-            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-            https://api.github.com/repos/${{ github.repository }}/pulls/$DRAINPIPE_PR_NUMBER \
-            -d @processed.json
+      - name: Post sticky pull request comment
+        if: ${{ steps.composer-lock-changed.outputs.any_changed == 'true' }}
+        uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
@deviantintegral suggests that we pin to a commit hash so future releases don't unexpectedly break our builds.

Reply:
How do we get notified of new releases to update the hash to? Monitor the log files for deprecation notices?

Reply:
While I agree that locking it down to commit hashes is better for security, the problem becomes when we need to update the workflows to all the consumers, or any repositories haven't been updated in a while, their actions could potentially break. It's a trade-off that we can balance with only doing it for very specific, trusted actions.